events/quarkus/src/main/java/org/projectnessie/events/quarkus/delivery/RetriableEventDelivery.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (C) 2023 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.events.quarkus.delivery;
+
+import java.time.Duration;
+import org.projectnessie.events.api.Event;
+import org.projectnessie.events.service.EventService;
+import org.projectnessie.events.spi.EventSubscriber;
+
+/**
+ * Abstract base class for event delivery implementations with retries and backoff.
+ *
+ * <p>The core delivery logic is implemented by {@link StandardEventDelivery}.
+ *
+ * <p>For blocking subscribers, {@link BlockingEventDelivery} should be used instead.
+ *
+ * <p>Other implementations use the delegate pattern to inject additional behavior: {@link
+ * LoggingEventDelivery}, {@link TracingEventDelivery}, and {@link MetricsEventDelivery}.
+ */
+public abstract class RetriableEventDelivery implements EventDelivery {
+
+  /**
+   * Starts the delivery.
+   *
+   * @implSpec Implementers must call {@link #startAttempt(int, Duration, Throwable)}.
+   */
+  @Override
+  public abstract void start();
+
+  /**
+   * Starts a new delivery attempt.
+   *
+   * @param currentAttempt The current delivery attempt, starting at 1.
+   * @param nextDelay The delay to wait before the next attempt, if this attempt fails.
+   * @param previousError The error from the previous attempt, or {@code null} if this is the first
+   *     attempt.
+   * @implSpec Implementers must call {@link #tryDeliver(int)}, then either {@link
+   *     #deliverySuccessful(int)} or, if the attempt failed, {@link #attemptFailed(int, Duration,
+   *     Throwable)}.
+   */
+  protected abstract void startAttempt(
+      int currentAttempt, Duration nextDelay, Throwable previousError);
+
+  /**
+   * Tries to deliver the event to the subscriber.
+   *
+   * @implSpec Implementers should call {@link EventService#notifySubscriber(EventSubscriber,
```

Comment:
This line (and the one below) should be:
```
   * @implSpec Implementers should call {@link EventSubscriber#onEvent(Event)}.
```
+ the import removed 

(didn't all work with a GH suggestion :( )